src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsLoggingBuilderExtensions.cs

```diff
@@ -61,8 +62,20 @@ public static class ApplicationInsightsLoggingBuilderExtensions
             builder.AddConfiguration();
             builder.Services.AddApplicationInsights(loggerOptionsConfiguration, additionalTelemetryConfiguration);
 
-            builder.Services.PostConfigure<LoggerFilterOptions>(o =>
-            {
+            builder.Services.AddOptions<LoggerFilterOptions>()
+                .PostConfigure<IOptions<ApplicationInsightsLoggerOptions>>((o, appInsightsOptions) =>
+                {
+                // The custom filtering below is only needed if we are sending all logs to Live Metrics.
+                // If we are filtering (or not using Live Metrics) we don't need to do this.
+                if (appInsightsOptions != null)
+                {
+                    if (!appInsightsOptions.Value.EnableLiveMetrics ||
+                        appInsightsOptions.Value.EnableLiveMetricsFilters)
```

Comment:
I think we need to tweak this... if `EnableLiveMetrics` is false, we should just immediately return, no matter what `EnableLiveMetricsFilters` is. I *think* in this case (you should add a configuration test) if `EnableLiveMetrics` is false and `EnableLiveMetricsFilters` is false... we'd still create this custom filter, when we really shouldn't.

Reply:
I feel we need the entire condition as it is.

_if EnableLiveMetrics is false, we should just immediately return, no matter what EnableLiveMetricsFilters is._

^ this is happening with current condition. If EnableLiveMetrics is false, first part of `||` will be always true and we will always return.

Second part of `||` is still needed for the scenario where EnableLiveMetrics is true but we want live metrics to be filtered (i.e. EnableLiveMetricsFilters  = true)

Let me know if I am missing anything.

Reply:
you're right :-) I was misreading!